src/Particle/ParticleSpatialLayout.h

```diff
@@ -73,11 +75,18 @@ namespace ippl {
         //! The RegionLayout which determines where our particles go.
         RegionLayout_t rlayout_m;
 
-        using region_type = typename RegionLayout_t::view_type::value_type;
+        //! The FieldLayout containing information on nearest neighbors
+        FieldLayout_t flayout_m;
+
+	    using region_view_type  = typename RegionLayout_t::view_type;
+        using region_type 	= typename region_view_type::value_type;
+	    using neighbor_list = typename FieldLayout_t::neighbor_list;
 
         template <size_t... Idx>
         KOKKOS_INLINE_FUNCTION constexpr static bool positionInRegion(
             const std::index_sequence<Idx...>&, const vector_type& pos, const region_type& region);
+        
+        size_type getNeighborSize(const neighbor_list& neighbors) const;
```

Comment:
add a comment what is the purpose of the function. use Doxygen. This is valid for other functions (and types)

Reply:
See latest commit (for all comments regarding doxygen)